courses/factories.py

```diff
@@ -7,56 +7,52 @@
 from factory import fuzzy
 from factory.django import DjangoModelFactory
 
-from .models import Program, Course, CourseRun
+from courses.models import Program, Course, CourseRun
+from ecommerce.models import CoursePrice
 
 FAKE = faker.Factory.create()
 
 
 class ProgramFactory(DjangoModelFactory):
     """Factory for Programs"""
-    title = fuzzy.FuzzyText(prefix="Program ")
-    live = fuzzy.FuzzyAttribute(FAKE.boolean)
-    description = fuzzy.FuzzyText()
+    title = factory.LazyAttribute(lambda x: FAKE.company())
+    description = factory.LazyAttribute(lambda x: FAKE.bs())
+    live = True
+    financial_aid_availability = False
 
-    class Meta:  # pylint: disable=missing-docstring
-        model = Program
+    course = factory.RelatedFactory('courses.factories.CourseFactory', "program")
 
-    @classmethod
-    def _create(cls, model_class, *args, **kwargs):
-        full_create = kwargs.pop('full', False)
-        program = model_class(*args, **kwargs)
-        program.save()
-        if full_create:
-            course = CourseFactory.create(program=program)
-            course_run = CourseRunFactory.create(course=course)
-            from ecommerce.factories import CoursePriceFactory
-            course_price = CoursePriceFactory.create(course_run=course_run, is_valid=True)
-            if program.financial_aid_availability:
-                from financialaid.factories import TierProgramFactory
+    @factory.post_generation
```

Comment:
see, this is what i _really_ wanted when i wrote that `full=True` nonsense :) thanks for finding this! i'll try to get to this PR today